### PR TITLE
Add tests for ArraySpeciesCreate

### PR DESCRIPTION
**Pull request by jugglinmike**
Assert the expected behavior of the ArraySpeciesCreate abstract
operation in the 5 Array instance methods from which it is invoked.

This change set does not include tests for ECMAScript realm
considerations because Test262 does not currently expose a mechanism for
interfacing with realms.

### CODE REVIEW

File: test/built-ins/Array/prototype/filter/create-species-non-ctor.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+       b. If C is null, let C be undefined.
+    [...]
+    9. If IsConstructor(C) is false, throw a TypeError exception.
+features: [Symbol.species]
+---*/
+
+var a = [];
+var callCount = 0;
+var cb = function() { callCount += 1; };
+var result;
```
Comment (goyakin) [inline test/built-ins/Array/prototype/filter/create-species-non-ctor.js]: `result` isn't needed.

**Comment by goyakin** [comment]
@jugglinmike A `result` variable is declared in a few test files, but they aren't actually used, so we should remove them. Everything else LGTM.

**Comment by jugglinmike** [comment]
Thanks for the review, Gorkem! I've removed all occurrence of unused variables (this includes `result` plus a few function parameters).

**Comment by jugglinmike** [comment]
Looks like I forgot to push those "fixup" commits (thanks for letting me know, @leobalter). This branch now includes the requested changes.

File: test/built-ins/Array/prototype/splice/create-species-neg-zero.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    4. If relativeStart < 0, let actualStart be max((len + relativeStart), 0);
+       else let actualStart be min(relativeStart, len).
+    5. If the number of actual arguments is 0, then
+       [...]
+    6. Else if the number of actual arguments is 1, then
+       [...]
+    7. Else,
+       a. Let insertCount be the number of actual arguments minus 2.
+       b. Let dc be ? ToInteger(deleteCount).
+       c. Let actualDeleteCount be min(max(dc, 0), len - actualStart).
```
Comment (leobalter) [inline test/built-ins/Array/prototype/splice/create-species-neg-zero.js]: This is just cosmetic, but it is cloudy for me. I believe `actualDeleteCount` becomes `+0` here on the max operation. In any way this is not observable before the step 2 of 9.4.2.3.

File: test/built-ins/Array/prototype/splice/create-species-neg-zero.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  args = arguments;
+};
+var a = [];
+a.constructor = {};
+a.constructor[Symbol.species] = Ctor;
+
+a.splice(0, -0);
+
+assert.sameValue(args.length, 1);
+assert.sameValue(args[0], 0);
```
Comment (leobalter) [inline test/built-ins/Array/prototype/splice/create-species-neg-zero.js]: and I always need to remind myself `sameValue` handles -0 and +0 as different values.

File: test/built-ins/Array/prototype/splice/create-species-neg-zero.js (hunk 2 above)
Comment (jugglinmike) [inline test/built-ins/Array/prototype/splice/create-species-neg-zero.js]: I disagree. The reason I included that step in this file is to justify the test. "max" does not distinguish between negative zero and positive zero, so the result of "max(0, -0)" is implementation-dependent.

Comment (leobalter) [inline test/built-ins/Array/prototype/splice/create-species-neg-zero.js]: > so the result of "max(0, -0)" is implementation-dependent.

I can't say I love the implementation-dependent parts, but it's reasonable. Thanks for the quick answer.

**Pull request merged**